### ISSUE TITLE
setuptools >= 67: pyparsing has been removed from pkg_resources

### ISSUE THREAD

**Issue by dimitry-unified-streaming** [py2exe/py2exe#177]
As mentioned in https://setuptools.pypa.io/en/stable/history.html#v67-0-0, via pypa/setuptools#3790, `pyparsing` has been removed from setuptool's vendored modules. So whenever you run py2exe it results in:

```text
running py2exe
Analyzing the code
Traceback (most recent call last):
  File "C:\build\make-exe.py", line 11, in <module>
    setuptools.setup(
  File "C:\Python311\Lib\site-packages\setuptools\__init__.py", line 108, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\setuptools\_distutils\core.py", line 185, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\setuptools\_distutils\core.py", line 201, in run_commands
    dist.run_commands()
  File "C:\Python311\Lib\site-packages\setuptools\_distutils\dist.py", line 969, in run_commands
    self.run_command(cmd)
  File "C:\Python311\Lib\site-packages\setuptools\dist.py", line 1213, in run_command
    super().run_command(command)
  File "C:\Python311\Lib\site-packages\setuptools\_distutils\dist.py", line 988, in run_command
    cmd_obj.run()
  File "C:\Python311\Lib\site-packages\py2exe\distutils_buildexe.py", line 204, in run
    self._run()
  File "C:\Python311\Lib\site-packages\py2exe\distutils_buildexe.py", line 284, in _run
    builder.analyze()
  File "C:\Python311\Lib\site-packages\py2exe\runtime.py", line 180, in analyze
    mf.import_package(modname)
  File "C:\Python311\Lib\site-packages\py2exe\mf310.py", line 144, in import_package
    self.import_hook(name,  None, ["*"])
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 215, in import_hook
    q, tail = self.find_head_package(parent, name)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 271, in find_head_package
    q = self.import_module(head, qname, parent)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 368, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 381, in load_module
    m = self.load_package(fqname, pathname)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 523, in load_package
    m = self.add_module(fqname, file=pathname, path=[pathname])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\mf310.py", line 130, in add_module
    self._add_module(fqname, m)
  File "C:\Python311\Lib\site-packages\py2exe\dllfinder.py", line 277, in _add_module
    self.hook(mod)
  File "C:\Python311\Lib\site-packages\py2exe\dllfinder.py", line 274, in hook
    mth(self, mod)
  File "C:\Python311\Lib\site-packages\py2exe\hooks.py", line 614, in hook_pkg_resources
    finder.import_package("pkg_resources._vendor.pyparsing")
  File "C:\Python311\Lib\site-packages\py2exe\mf310.py", line 144, in import_package
    self.import_hook(name,  None, ["*"])
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 216, in import_hook
    m = self.load_tail(q, tail)
        ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Python311\Lib\site-packages\py2exe\vendor\modulefinder.py", line 299, in load_tail
    raise ImportError("No module named " + mname)
ImportError: No module named pkg_resources._vendor.pyparsing
```

Possibly there should be a version check in `hook_pkg_resources` ?

**Comment by albertosottile** [comment]
I wanted to work on "persistent" fix for this, but it looks like none of the tests in the CI can reproduce this (perhaps they do not require explicitly `pkg_resources` or any of its `extern` dependencies).

Can you provide a minimal example that can be used to test this bug? Thanks

**Comment by albertosottile** [comment]
I tried to write a test on my own and I attempted to fix this bug in the `develop` branch. You may find preliminary wheels that include the fix here: https://github.com/py2exe/py2exe/actions/runs/4157014939

@dimitry-unified-streaming Could you try again with these wheels and check if you can still reproduce the bug? Thanks

**Comment by albertosottile** [comment]
Fixed in 0.13.0.1
